### ISSUE TITLE
Crunchyroll format selects hardsub-ptBR?

### ISSUE THREAD

**Issue by erickyun** [yt-dlp/yt-dlp#172]
[info] Downloading format(s) adaptive_hls-audio-jaJP-hardsub-ptBR-8112-1

İt always select prBR when -f best is used how can I get adaptive_hls-audio-jaJP-8112-1 without hardsub subtitles as default?

Also extracting formats is slow İt takes a while before it start downloading?

**Comment by pukkandan** [comment]
Give the output of `yt-dlp -vF URL`

**Comment by erickyun** [comment]
> Give the output of `yt-dlp -vF URL`

[debug] Command-line config: ['-vF', 'CrURL', '--hls-use-mpegts', '-f', 'NAME', '--cookies', 'cookies.txt', '--sub-lang', 'enUS', '--write-sub', '--force-ipv4', '--http-chunk-size', '10M', '--external-downloader', 'aria2c', '--external-downloader-args', 'aria2c:-x 16 -s 16 -k 10M --always-resume false --max-resume-failure-tries=1 --max-tries=1000000000 --log-level=error --log=log.txt']                                            [debug] Loading archive file None
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8                                                [debug] yt-dlp version 2021.03.15
[debug] Python version 3.8.5 (CPython 64bit) - Linux-5.4.0-fake-kernel-aarch64-with-glibc2.29
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4, rtmpdump 2.4
[debug] Proxy map: {}

**Comment by pukkandan** [comment]
@erickyun Entire output. I want to see the formats

**Comment by erickyun** [comment]
> @erickyun Entire output. I want to see the formats

[debug] Proxy map: {}
[crunchyroll] 801388: Downloading webpage
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-ptBR information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-esES information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-arME information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-esLA information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-deDE information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-frFR information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-enUS information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-ruRU information
[crunchyroll] 801388: Downloading adaptive_hls-audio-jaJP-hardsub-itIT information
[debug] Sort order given by user:
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[crunchyroll] 801388: Downloading media info
[info] Available formats for 801388:
ID                                          EXT RESOLUTION FPS |   TBR PROTO | VCODEC        VBR ACODEC     ABR NOTE
------------------------------------------- --- ---------- --- - ----- ----- - ----------- ----- --------- ---- ------
adaptive_hls-audio-jaJP-hardsub-frFR-561-0  mp4 428x240    23  |  561k m3u8  | avc1.42c015  561k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-561-1  mp4 428x240    23  |  561k m3u8  | avc1.42c015  561k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-561-0  mp4 428x240    23  |  561k m3u8  | avc1.42c015  561k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-561-1  mp4 428x240    23  |  561k m3u8  | avc1.42c015  561k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-561-0  mp4 428x240    23  |  561k m3u8  | avc1.42c015  561k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-561-1  mp4 428x240    23  |  561k m3u8  | avc1.42c015  561k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-561-0  mp4 428x240    23  |  561k m3u8  | avc1.42c015  561k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-561-1  mp4 428x240    23  |  561k m3u8  | avc1.42c015  561k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-562-0               mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-562-1               mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-562-0  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-562-1  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-562-0  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-562-1  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-562-0  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-562-1  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-562-0  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-562-1  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-562-0  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-562-1  mp4 428x240    23  |  562k m3u8  | avc1.42c015  562k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-1091-0              mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-1091-1              mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-1091-0 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-1091-1 mp4 640x360    23  | 1091k m3u8  | avc1.4d401e 1091k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-2089-0              mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-2089-1              mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-2089-0 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-2089-1 mp4 848x480    23  | 2089k m3u8  | avc1.4d401f 2089k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-4106-0 mp4 1280x720   23  | 4106k m3u8  | avc1.640028 4106k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-4106-1 mp4 1280x720   23  | 4106k m3u8  | avc1.640028 4106k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-4106-0 mp4 1280x720   23  | 4106k m3u8  | avc1.640028 4106k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-4106-1 mp4 1280x720   23  | 4106k m3u8  | avc1.640028 4106k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-4106-0 mp4 1280x720   23  | 4106k m3u8  | avc1.640028 4106k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-4106-1 mp4 1280x720   23  | 4106k m3u8  | avc1.640028 4106k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-4106-0 mp4 1280x720   23  | 4106k m3u8  | avc1.640028 4106k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-4106-1 mp4 1280x720   23  | 4106k m3u8  | avc1.640028 4106k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-4107-0 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-4107-1 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-4107-0 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-4107-1 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-4107-0 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-4107-1 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-4107-0              mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-4107-1              mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-4107-0 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-4107-1 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-4107-0 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-4107-1 mp4 1280x720   23  | 4107k m3u8  | avc1.640028 4107k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-8110-0 mp4 1920x1080  23  | 8110k m3u8  | avc1.640028 8110k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-itIT-8110-1 mp4 1920x1080  23  | 8110k m3u8  | avc1.640028 8110k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-8111-0 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-enUS-8111-1 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-8111-0 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ruRU-8111-1 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-8111-0 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-arME-8111-1 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-8111-0 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esLA-8111-1 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-8111-0              mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-8111-1              mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-8111-0 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-deDE-8111-1 mp4 1920x1080  23  | 8111k m3u8  | avc1.640028 8111k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-8112-0 mp4 1920x1080  23  | 8112k m3u8  | avc1.640028 8112k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-frFR-8112-1 mp4 1920x1080  23  | 8112k m3u8  | avc1.640028 8112k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-8112-0 mp4 1920x1080  23  | 8112k m3u8  | avc1.640028 8112k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-esES-8112-1 mp4 1920x1080  23  | 8112k m3u8  | avc1.640028 8112k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-8112-0 mp4 1920x1080  23  | 8112k m3u8  | avc1.640028 8112k mp4a.40.2   0k [jaJP]
adaptive_hls-audio-jaJP-hardsub-ptBR-8112-1 mp4 1920x1080  23  | 8112k m3u8  | avc1.640028 8112k mp4a.40.2   0k [jaJP]
root@localhost:/storage/emulated/0/download/new#

**Comment by pukkandan** [comment]
It was a bug in the format sorting code. I have fixed it. Until the next release, you use `-f "(bv*+ba/b)[format_id!*=hardsub]"` as a temporary workaround

**Comment by erickyun** [comment]
> It was a bug in the format sorting code. I have fixed it. Until the next release, you use `-f "(bv*+ba/b)[format_id!*=hardsub]"` as a temporary workaround

Thank you very much🙂

İt was also slow starting when command was executed before the download. İs this also fixed?

**Comment by pukkandan** [comment]
> İt was also slow starting when command was executed before the download

How slow? it shouldn't take more than a few seconds on a good network

**Comment by erickyun** [comment]
> > İt was also slow starting when command was executed before the download
> 
> How slow? it shouldn't take more than a few seconds on a good network

Slower than youtube-dl
More than 40sec ?
My network is good 10MiB/s

**Comment by pukkandan** [comment]
> Also speed is normal for pc but not termux android?

How are you installing on termux? pip, or provided binary?

> Did try -f "(bv*+ba/b)[format_id!*=hardsub]" İt isnt working on termux but it does work on windows 10 pc?

Try with single quotes on termux: `-f '(bv*+ba/b)[format_id!*=hardsub]'`

**Comment by erickyun** [comment]
> > Also speed is normal for pc but not termux android?
> 
> How are you installing on termux? pip, or provided binary?
> 
> > Did try -f "(bv*+ba/b)[format_id!*=hardsub]" İt isnt working on termux but it does work on windows 10 pc?
> 
> Try with single quotes on termux: `-f '(bv*+ba/b)[format_id!*=hardsub]'`

Yes single quotes does work thank you.🙂👍

**Comment by erickyun** [comment]
I noticed another bug btw after it finished downloading.
yt-dlp deletes .ass subtitle?

**Comment by pukkandan** [comment]
again, show full verbose log.

Actually, open a new issue since its unelated to the original one

**Comment by erickyun** [comment]
> again, show full verbose log.
> 
> Actually, open a new issue since its unelated to the original one

Sorry for bothering you . I did solved it. I did removed --always-resume false

**Comment by pukkandan** [comment]
> with ubuntu installing on termux and then with pip3 ı download yt-dlp

I still don't understand how you installed it
If you are using PIP, the command you should have used is `pip3 install --upgrade yt-dlp`
See the [installation instructions](https://github.com/yt-dlp/yt-dlp#installation)

**Comment by erickyun** [comment]
> > with ubuntu installing on termux and then with pip3 ı download yt-dlp
> 
> I still don't understand how you installed it
> If you are using PIP, the command you should have used is `pip3 install --upgrade yt-dlp`
> See the [installation instructions](https://github.com/yt-dlp/yt-dlp#installation)

Sorry for mine bad English that is what I mean that is how I installed.

But the problem is not only on termux?
I tested on mine window10 pc and saw the same problem! İt launching starting slow before starting to download?

**Comment by pukkandan** [comment]
starting slow as compared to a similarly installed youtube-dl? The first run of the program will be slow, but consecutive runs should not be. Can you show me the output of the following 2 commands (on windows) to confirm.

    yt-dlp & echo.%time% & yt-dlp -v & echo.%time%
    youtube-dl & echo.%time% & youtube-dl -v & echo.%time%

**Comment by erickyun** [comment]
> starting slow as compared to a similarly installed youtube-dl? The first run of the program will be slow, but consecutive runs should not be. Can you show me the output of the following 2 commands (on windows) to confirm.
> 
> ```
> yt-dlp & echo.%time% & yt-dlp -v & echo.%time%
> youtube-dl & echo.%time% & youtube-dl -v & echo.%time%
> ```

the command isn't working 


PS C:\Users\aydin\Desktop\yt-dl\Aria2> yt-dlp & echo.%time% & yt-dlp -v & echo.%time%
At line:1 char:8
+ yt-dlp & echo.%time% & yt-dlp -v & echo.%time%
+        ~
The ampersand (&) character is not allowed. The & operator is reserved for future use; wrap an ampersand in double quotation marks ("
&") to pass it as part of a string.
At line:1 char:22
+ yt-dlp & echo.%time% & yt-dlp -v & echo.%time%
+                      ~
The ampersand (&) character is not allowed. The & operator is reserved for future use; wrap an ampersand in double quotation marks ("
&") to pass it as part of a string.
At line:1 char:34
+ yt-dlp & echo.%time% & yt-dlp -v & echo.%time%
+                                  ~
The ampersand (&) character is not allowed. The & operator is reserved for future use; wrap an ampersand in double quotation marks ("
&") to pass it as part of a string.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : AmpersandNotAllowed

**Comment by pukkandan** [comment]
Run it in cmd, not powershell

**Comment by erickyun** [comment]
> Run it in cmd, not powershell

## sorry my bad

C:\Users\aydin>yt-dlp & echo.%time% & yt-dlp -v & echo.%time%
Usage: yt-dlp [OPTIONS] URL [URL...]

yt-dlp: error: You must provide at least one URL.
Type yt-dlp --help to see a list of all options.
 9:33:24,89
[debug] Command-line config: ['-v']
[debug] Loading archive file None
[debug] Encodings: locale cp1254, fs utf-8, out utf-8, pref cp1254
[debug] yt-dlp version 2021.03.15 (exe)
[debug] Python version 3.8.7 (CPython 64bit) - Windows-10-10.0.18362-SP0
[debug] exe versions: ffmpeg 4.3.1-2020-10-01-full_build-www.gyan.dev, ffprobe 4.3.1-2020-10-01-full_build-www.gyan.dev
[debug] Proxy map: {}
Usage: yt-dlp [OPTIONS] URL [URL...]

yt-dlp: error: You must provide at least one URL.
Type yt-dlp --help to see a list of all options.
 9:33:24,89

C:\Users\aydin>youtube-dl & echo.%time% & youtube-dl -v & echo.%time%
Usage: youtube-dl [OPTIONS] URL [URL...]

youtube-dl: error: You must provide at least one URL.
Type youtube-dl --help to see a list of all options.
 9:34:30,14
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v']
[debug] Encodings: locale cp1254, fs utf-8, out utf-8, pref cp1254
[debug] youtube-dl version 2021.03.03
[debug] Python version 3.9.1 (CPython) - Windows-10-10.0.18362-SP0
[debug] exe versions: ffmpeg 4.3.1-2020-10-01-full_build-www.gyan.dev, ffprobe 4.3.1-2020-10-01-full_build-www.gyan.dev
[debug] Proxy map: {}
Usage: youtube-dl [OPTIONS] URL [URL...]

youtube-dl: error: You must provide at least one URL.
Type youtube-dl --help to see a list of all options.
 9:34:30,14

**Comment by pukkandan** [comment]
oops, sorry. I forgot that windows cmd does not do delayed variable expansion 🤦
 So that command doesn't work like I thought 😢 

But I was able to confirm from my end that the issue does exist. I'm opening a new issue for this

**Comment by pukkandan** [comment]
actually, nevermind. Once **I turned off the antivirus**, `yt-dlp` ended up being faster than `youtube-dl`, (though not by much).

```
$ youtube-dl -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2021.03.14
[debug] Python version 3.4.4 (CPython) - Windows-7-6.1.7601-SP1
[debug] exe versions: none
[debug] Proxy map: {}
Usage: youtube-dl [OPTIONS] URL [URL...]

youtube-dl: error: You must provide at least one URL.
Type youtube-dl --help to see a list of all options.

$ yt-dlp -v
[debug] Command-line config: ['-v']
[debug] Loading archive file None
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.03.15 (exe)
[debug] Python version 3.8.7 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] exe versions: none
[debug] Proxy map: {}
Usage: yt-dlp [OPTIONS] URL [URL...]

yt-dlp: error: You must provide at least one URL.
Type yt-dlp --help to see a list of all options.

$ time-cmd youtube-dl
Usage: youtube-dl [OPTIONS] URL [URL...]

youtube-dl: error: You must provide at least one URL.
Type youtube-dl --help to see a list of all options.

Completed in 7 seconds

$ time-cmd yt-dlp
Usage: yt-dlp [OPTIONS] URL [URL...]

yt-dlp: error: You must provide at least one URL.
Type yt-dlp --help to see a list of all options.

Completed in 6 seconds

$ time-cmd youtube-dl --skip-download ytsearch:test
[download] Downloading playlist: test
[youtube:search] query "test": Downloading page 1
[youtube:search] playlist test: Downloading 1 videos
[download] Downloading video 1 of 1
[youtube] M4j4dp-Pj0o: Downloading webpage
[download] Finished downloading playlist: test

Completed in 14 seconds

$ time-cmd yt-dlp --skip-download ytsearch:test
[download] Downloading playlist: test
[youtube:search] query "test": Downloading page 1
[youtube:search] playlist test: Downloading 1 videos
[download] Downloading video 1 of 1
[youtube] M4j4dp-Pj0o: Downloading webpage
[info] Downloading format(s) 22
[download] Finished downloading playlist: test

Completed in 13 seconds
```

**Comment by erickyun** [comment]
> actually, nevermind. Once **I turned off the antivirus**, `yt-dlp` ended up being faster than `youtube-dl`, (though not by much).

For me the issue is still there?
And I have the Same problem for vrv.co too!

**Comment by pukkandan** [comment]
Is the issue only for these 2 websites? I thought you were talking about a general issue. **Please open a new issue** with the relevent details
